embed.rkt

```diff
@@ -713,16 +714,21 @@ GIF XMP keyword: #"XMP Data" with auth #"XMP"
                      ([tag (in-list lst)])
              (append bag `((rdf:li () ,tag)))))))
 
+; fixes issues when we have multiple elements inside a single rdf:li
+(define/contract (rdf:li-fixer rdf:li)
+  ((listof is-rdf:li?) . -> . list?)
+  (for/fold ([lst empty])
```

Comment:
So if we're iterating over a list, and outputting a new list that's the same length, that is starting to look a lot like a candidate for `map` instead of `for/fold`. Then we don't have to manage an explicit accumulator or worry about manually appending anything.

Consider the following snippet, which is pretty artificial, but incorporates all the same basic moving parts as `rdf:li-fixer`:

```racket
#lang racket

(map (λ (elem)
       (if (> (length elem) 1)
           (apply string-append elem)
           elem))
     (map (λ (num) (list "0" (number->string num)))  (list 1 2 3 4)))
```